utils/zerovec/src/ule/plain.rs

```diff
@@ -364,3 +364,36 @@ impl AsULE for bool {
 
 // EqULE is true because bool is its own ULE.
 unsafe impl EqULE for bool {}
+
+// Safety (based on the safety checklist on the ULE trait):
+//  1. () does not include any uninitialized or padding bytes (it has no bytes)
+//  2. () is aligned to any number of bytes (it is zero-sized)
+//  3. The impl of validate_bytes() returns an error if any byte is not valid (any byte).
+//  4. The impl of validate_bytes() returns an error if there are leftover bytes (always).
+//  5. The other ULE methods use the default impl.
+//  6. () byte equality is semantic equality
+unsafe impl ULE for () {
```

Comment:
This is potentially subtly unsound, ZSTs are weird.

https://github.com/rust-lang/unsafe-code-guidelines/issues/557

Reply:
It is not unsound, but let's include a link to that issue.

Reply:
Thanks for investigating.

Can I ask you to make a Suggested Edit? I'm not completely sure I like my unsafe docs, especially the thing about being align(1).

Reply:
I removed this impl but added a new impl and modified the safety constraint. PTAL.

Reply:
keeping safety discussion in one place

Reply:
I'll be sure to provide safety docs once we've settled on an approach